test/csuite/timestamp_abort/main.c

```diff
@@ -458,8 +452,17 @@ thread_run(void *arg)
             testutil_check(prepared_session->begin_transaction(prepared_session, NULL));
 
         if (use_ts) {
-            /* Allocate two timestamps. */
-            active_ts = __wt_atomic_fetch_addv64(&global_ts, 2);
+            /*
+             * Our timestamps are derived from which key we are writing, in relation to the entire
+             * set of keys being written. In a given iteration for each thread, there are three
+             * timestamps available, though we don't always use the third. The first is used to
+             * timestamp the transaction at the beginning. The second is used to timestamp after an
+             * insert is done. Then, we sometimes want the durable timestamp ahead of the commit
+             * timestamp, so we reserve the last timestamp for that use.
+             *
+             * Set the active timestamp to the first of the three timestamps we'll use.
+             */
+            active_ts = 1 + 3 * (td->info + i * nth);
```

Comment:
Is the "there are three timestamps available" the same as the `3 * (expr)` in the code? Does it really mean we allocate three timestamps on each iteration? Maybe a slight rewording to be:
In a given iteration, allocate three timestamps for the thread to use on this iteration. The first... The second... Then, we sometimes... so we reserve the third timestamp for that use.

I guess "available" and "allocated" have slightly different meanings to me and it seems we're allocating them.

Reply:
Yes, the 3* relates to three timestamps.  I can use "allocated" if that makes sense - to be clear, everything is preallocated, there is no actual allocation happening in this statement.  It's like of like saying (if you had two threads): "I'll take the even numbers, you take the odds".  So that "allocates" the set of integers between two threads.  Once we decide, it's just a simple computation to decide the next number.  Should I expand the comment, or is it clear enough as is (with the change to "allocated")?

Reply:
Thanks. So it is a reserved timestamp space for each thread. Reading that expression makes me need to parse, process and reason about what it is doing in the loop. Maybe add a macro that is `MY_NEXT_TS` and have a comment about each thread's reserved timestamp space.

Reply:
Thanks, I've added a macro `RESERVED_TIMESTAMPS_FOR_ITERATION` and put the big comment above it.  I also rearranged the arithmetic so maybe it's a little easier to grok.